### PR TITLE
ADD python-rich as required by Netplan

### PR DESCRIPTION
**Pull request by ndubchak**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [ ] The toolchain has been rebuilt successfully (or no changes were made to it)
- [ ] The toolchain/worker package manifests are up-to-date
- [ ] Any updated packages successfully build (or no packages were changed)
- [ ] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [ ] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [ ] All package sources are available
- [ ] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [ ] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [ ] All source files have up-to-date hashes in the `*.signatures.json` files
- [ ] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [ ] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

This PR adds a new package to 3.0 called python-rich. This package is required as a dependency for Netplan.

###### Change Log  

- Added a new package python-rich

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Pipeline build id: xxxx

### CODE REVIEW

File: SPECS/python-rich/python-rich.spec [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+# add below to make sure initial build will catch runtime import errors
+%pyproject_check_import
+%pytest -vv
+ 
+%files -n python3-rich -f %{pyproject_files}
+%license LICENSE
+%doc README.md
+ 
+%changelog
+* Mon Mar 18 2024 Nadiia Dubchak <ndubchak@microsoft.com> - 13.7.1
```
Comment (vinceaperri) [inline SPECS/python-rich/python-rich.spec:63]:
```
* Mon Mar 18 2024 Nadiia Dubchak <ndubchak@microsoft.com> - 13.7.1-2
- Initial CBL-Mariner import from Fedora 41 (license: MIT).
- License verified.
```

File: SPECS/python-rich/python-rich.spec [hunk 2]
```diff
@@ -0,0 +1,175 @@
+Name:           python-rich
+Version:        13.7.1
+Release:        1%{?dist}
```
Comment (vinceaperri) [inline SPECS/python-rich/python-rich.spec:3]: Release should be bumped to distinguish it from previous fedora entry in the changelog.

File: SPECS/poetry/poetry.spec
Comment (vinceaperri) [inline SPECS/poetry/poetry.spec:1]: Poetry looks to be very out of date. Should be upgraded to latest stable (1.8). Fedora 40 is at 1.7.

**Review by vinceaperri — CHANGES_REQUESTED** [review]

File: SPECS/python-rich/python-rich.spec (hunk 1 above)
Comment (vinceaperri) [reply SPECS/python-rich/python-rich.spec:63]: I think putting License verified. on its own line will make the license check pass.

Comment (vinceaperri) [reply SPECS/python-rich/python-rich.spec:63]: Ah, never mind, the error is for poetry. May need to add "License verified" to that too.

File: SPECS/poetry/poetry.spec [hunk 3]
```diff
@@ -1,75 +1,96 @@
...
+URL:            https://python-poetry.org/
+Source0:        https://github.com/python-poetry/poetry/archive/%{version}/poetry-%{version}.tar.gz
 
 # relax some too-strict dependencies that are specified in setup.py:
 # - importlib-metadata (either removed or too old in fedora)
 # - keyring (too new in fedora, but should be compatible)
 # - pyrsistent (too new in fedora, but should be compatible)
 # - requests-toolbelt (too new in fedora, but should be compatible)
-Patch0:         00-setup-requirements-fixes.patch
+# Patch0:         00-setup-requirements-fixes.patch
```
Comment (Redent0r) [inline SPECS/poetry/poetry.spec:22]: May remove this line and patch file if it no longer applies

**Pull request closed without merging**